src/test/java/org/kainos/ea/service/JobServiceUnitTest.java

```diff
@@ -59,9 +60,37 @@ void getAllJobs_shouldThrowFailedToGetJobsException_whenDaoThrowsSQLException()
     @Test
     void getAllJobs_shouldThrowFailedToGetJobsException_whenDatabaseConnectorThrowsSQLException() throws SQLException {
         Mockito.when(databaseConnector.getConnection()).thenThrow(SQLException.class);
+        Mockito.when(jobDao.getJobById(connection,-1)).thenReturn(null);
 
         assertThrows(FailedToGetJobsException.class,
                 () -> jobService.getAllJobs());
     }
 
+    @Test
+    void deleteJob_shouldReturnFalse_IfJobIdNotFound() throws SQLException, FailedtoDeleteException {
+        Mockito.when(databaseConnector.getConnection()).thenReturn(connection);
+        Mockito.when(jobDao.getJobById(connection,-1)).thenReturn(null);
+        Mockito.doNothing().when(jobDao).deleteJob(isA(Connection.class),isA(Integer.class));
+        assertFalse(jobService.delete(-1));
+    }
+
+    @Test
+    void deleteJob_shouldReturnTrue_IfJobIdFound() throws SQLException, FailedtoDeleteException {
+        Job sampleJob1 = new Job(1, "Software Engineer","Capbility", "Develops, tests, and maintains software applications, collaborates with cross-functional teams, and follows best practices to deliver efficient and reliable software solutions.", "https://kainossoftwareltd.sharepoint.com/people/Job%20Specifications/Forms/AllItems.aspx?id=%2Fpeople%2FJob%20Specifications%2FEngineering%2FJob%20profile%20%2D%20Software%20Engineer%20%28Associate%29%2Epdf&parent=%2Fpeople%2FJob%20Specifications%2FEngineering&p=true&ga=1","Trainee");
+        Mockito.when(databaseConnector.getConnection()).thenReturn(connection);
+        Mockito.when(jobDao.getJobById(connection,-1)).thenReturn(sampleJob1);
+        Mockito.doNothing().when(jobDao).deleteJob(isA(Connection.class),isA(Integer.class));
+        assertFalse(jobService.delete(1));
```

Comment:
this asserts false which isn't what the method name suggests

Reply:
Changed to check for nothing being thrown to align with new method behaviour.